deepchem/feat/base_classes.py

```diff
@@ -73,27 +121,58 @@ def _featurize_complex(self, mol_pdb, complex_pdb):
     raise NotImplementedError('Featurizer is not defined.')
 
 
-class Featurizer(object):
-  """
-  Abstract class for calculating a set of features for a molecule.
+class MolecularFeaturizer(Featurizer):
+  """Abstract class for calculating a set of features for a
+  molecule.
+
+  The defining feature of a `MolecularFeaturizer` is that it
+  uses SMILES strings and RDKIT molecule objects to represent
+  small molecules. All other featurizers which are subclasses of
+  this class should plan to process input which comes as smiles
+  strings or RDKIT molecules. 
 
-  Child classes implement the _featurize method for calculating features
-  for a single molecule.
+  Child classes need to implement the _featurize method for
+  calculating features for a single molecule.
+
+  Note
+  ----
+  In general, subclasses of this class will require RDKit to be installed.
   """
 
-  def featurize(self, mols, verbose=True, log_every_n=1000):
-    """
-    Calculate features for molecules.
+  def featurize(self, mols, log_every_n=1000):
+    """Calculate features for molecules.
 
     Parameters
     ----------
-    mols : iterable
-        RDKit Mol objects.
+    mols : RDKit Mol / SMILES string /iterable
```

Comment:
I think there is a possibility to misunderstand that `mols` takes only RDKit Mol.

`mol_datapoints` is better...?  it's better to imply that this variable could also take SMILES